### PR TITLE
feat: adding ruby3.3 image

### PR DESCRIPTION
**Pull request by hawflau**
*Issue #, if available:*

*Description of changes:*
Adding SAM Build Image for Ruby3.3 based on https://gallery.ecr.aws/lambda/ruby

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: build-image-src/Dockerfile-ruby33 [hunk 1]
```diff
@@ -0,0 +1,59 @@
+ARG IMAGE_ARCH
+# Update(hawflau): remove preview tag
+FROM public.ecr.aws/lambda/ruby:3.3-preview-$IMAGE_ARCH
```
Comment (mndeveci) [inline build-image-src/Dockerfile-ruby33]: - [x] Remove preview tag

**Review by mndeveci — APPROVED** [review]
LGTM! Trying to see if tasks can help us to remember removing the preview tag 🤔

**Review by mildaniel — APPROVED** [review]

File: build-image-src/Dockerfile-ruby33 (hunk 1 above)
Comment (hawflau) [inline build-image-src/Dockerfile-ruby33]:
```suggestion
FROM public.ecr.aws/lambda/ruby:3.3-$IMAGE_ARCH
```

File: build-image-src/Dockerfile-ruby33 [hunk 2]
```diff
@@ -0,0 +1,59 @@
+ARG IMAGE_ARCH
+# Update(hawflau): remove preview tag
```
Comment (hawflau) [inline build-image-src/Dockerfile-ruby33]:
```suggestion
```

**Pull request merged**